### PR TITLE
Improved translator performances

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Improved performances of translator in both front and back office
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Take a look at translations in front office, back office and specifically in the Translation Back Office 

=> [Blackfire comparison](https://blackfire.io/profiles/compare/491a1e28-616e-4d34-947f-3a70c284ab98/graph) <=

bouuum

### CODE REVIEW

File: classes/Context.php [hunk 1]
```diff
@@ -352,13 +361,13 @@ public function getTranslator()
             }
 
             $this->translator->addLoader('db', $sqlTranslationLoader);
+            $notName = $adminContext ? 'Shop*' : 'Admin*';
```
Comment (aleeks) [inline classes/Context.php]: Easu, but ultra smart! :scream:

**Comment by mickaelandrieu** [comment]
@basma-yangui-prestashop this is a very huge performance improvement, but this need to be tested.

3 common tests:
* check translations in front office
* check translations in back office
* in translation back office part:
 => do we still have the same behavior as in develop?
 => can we access the complete catalogue ? i.e are "Admin", "Shop" domains availables ?

File: classes/Context.php [hunk 2]
```diff
@@ -352,13 +361,13 @@ public function getTranslator()
...
 
             $finder = Finder::create()
                 ->files()
-                ->filter(function (\SplFileInfo $file) {
-                    return 2 === substr_count($file->getBasename(), '.') && preg_match('/\.\w+$/', $file->getBasename());
-                })
-                ->in($this->getTranslationResourcesDirectories())
+                ->name('*.*.xlf')
+                ->notName($notName)
+                ->in($this->getTranslationResourcesDirectories($adminContext))
```
Comment (mickaelandrieu) [inline classes/Context.php]:
```diff
- ->in($this->getTranslationResourcesDirectories($adminContext))
+ ->in($this->getTranslationResourcesDirectories()
```

Comment (aleeks) [inline classes/Context.php]: self reviewing, :tada:

Comment (mickaelandrieu) [inline classes/Context.php]: I'm not easy with myself :D

**Comment by basma-yangui-prestashop** [comment]
@mickaelandrieu When installing PS in english and add the french language, there are many fields which are not correctly translated in french. Same thing if I install in french and then add the english language (many fields aren't correctly translated to english)
Example in screenshot attached :
capture d ecran de 2017-01-09 14 32 38

**Comment by mickaelandrieu** [comment]
@basma-yangui-prestashop thank for your feedback, I found the bug and fixed it: what do you think ?

**Comment by basma-yangui-prestashop** [comment]
@mickaelandrieu Unfortunately, I still find some untranslated fields
capture d ecran de 2017-01-10 12 45 31
capture d ecran de 2017-01-10 12 22 26

**Comment by mickaelandrieu** [comment]
As it's not really related to this PR, I suggest we lock it until ALL translations bugs are fixed in Front office.

**Comment by vincentbz** [comment]
If some strings are translated before the PR and are back in english after the PR, can we guess it's related ? for example the "you may unsubscribe at any moment..." that is in FR before and in EN after the PR

**Comment by maximebiloe** [comment]
Thank you @mickaelandrieu!

**Pull request merged**